fix(boards): add endpoint to get most recently elected boards

LeonVreling opened the pull request:
Needed for functioning of [frontend#2036](https://github.com/AEGEE/frontend/pull/2036)

LeonVreling committed:
feat(boards): Add endpoint to get all current boards

WikiRik commented:
Mentioning it here, but also holds for the other PRs (and will include it on mine in the coming days); would be good if we add some tests. I want to rewrite our testing framework in relation to the mocking of core, but in the meantime we should still add tests

LeonVreling committed:
feat(boards): Add endpoint to get all elections within last year

This is the actual endpoint that I needed as for the AC check, and not the one from the previous commit, but we still might want to use that one for something

LeonVreling committed:
feat(boards): Add test for getting most recently elected boards

LeonVreling committed:
fix(boards): Remove need for end date recently elected boards

Also changed an endpoint to make it a more logical name

LeonVreling committed:
chore(boards): Fix lint

LeonVreling committed:
chore(boards): Hopefully fix new tests

LeonVreling committed:
chore(boards): Actually rename endpoint everywhere, and not just once

LeonVreling committed:
chore(boards): Fix formatting date string

LeonVreling committed:
Merge branch 'stable' into get-all-current-boards

LeonVreling committed:
Merge branch 'stable' into get-all-current-boards

WikiRik commented on test/api/board-listing.test.js:
```diff
@@ -37,11 +37,24 @@ describe('Board listing', () => {
     });
 
-    test('should fail for the current board of body if no permission', async () => {
+    test('should fail for most recent elected board of all bodies if no permission', async () => {
         mock.mockAll({ mainPermissions: { noPermissions: true } });
 
         const res = await request({
-            uri: '/bodies/1/boards/current',
```
Why remove the already existing test?

LeonVreling committed:
chore(network): Update before merging

Remove commented out code and add back previously deleted test

LeonVreling committed:
Merge remote-tracking branch 'origin/get-all-current-boards' into get-all-current-boards

serge1peshcoff commented:
:tada: This PR is included in version 1.2.1 :tada:

The release is available on [GitHub release](https://github.com/AEGEE/network/releases/tag/1.2.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
